controllers/authentication/utils.go

```diff
@@ -5,11 +5,51 @@
 package authentication
 
 import (
+	"encoding/json"
 	"errors"
+	"fmt"
 
-	"gopkg.in/square/go-jose.v2/jwt"
+	jwt "gopkg.in/square/go-jose.v2/jwt"
 )
 
+func extractClaims(tokenStr string, extraClaims []string) (map[string][]string, error) {
+	extra := make(map[string][]string)
+
+	if len(extraClaims) == 0 {
+		return extra, nil
+	}
+
+	token, err := jwt.ParseSigned(tokenStr)
+	if err != nil {
+		return extra, errors.New("cannot parse jwt token")
```

Comment:
Please return `nil` when also error is returned. Also for the other occurrences in this function.
```suggestion
		return nil, errors.New("cannot parse jwt token")
```

Reply:
Done